{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "650f751d",
   "metadata": {},
   "source": [
    "Ambiente miniconda modeller"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ec48ead7",
   "metadata": {},
   "outputs": [],
   "source": [
    "from modeller import *\n",
    "from modeller.automodel import *\n",
    "import os.path\n",
    "from pathlib import Path\n",
    "from Bio import SeqIO\n",
    "from Bio.PDB.PDBIO import Select\n",
    "from Bio.PDB import PDBParser, PPBuilder\n",
    "from Bio.PDB import PDBIO\n",
    "from io import StringIO\n",
    "import os\n",
    "import subprocess\n",
    "import requests\n",
    "import urllib\n",
    "import sys\n",
    "import urllib.request"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd4e94c8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4zwb.pdb ya existe. Abortando descarga.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'../run/modelamineto_dinamica/4zwb.pdb'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#=== Descarga de archivo PDB (templado) ===\n",
    "\n",
    "template = '4zwb'\n",
    "uniprotcd= 'P11166'\n",
    "\n",
    "def download_pdb(pdbid, url=\"https://files.rcsb.org/download/\"):\n",
    "    targetdir = \"../run/modelamineto_dinamica\"\n",
    "    os.makedirs(targetdir, exist_ok=True)\n",
    "    \n",
    "    pdbfullname = pdbid + \".pdb\"\n",
    "    # Modificar outfnm para incluir la ruta de destino\n",
    "    out = os.path.join(targetdir, pdbfullname)\n",
    "\n",
    "    if os.path.exists(out):\n",
    "        print(f\"{pdbfullname} ya existe. Abortando descarga.\")\n",
    "        return out\n",
    "    else:\n",
    "        url = url + pdbfullname\n",
    "        try:\n",
    "            urllib.request.urlretrieve(url, out)\n",
    "            print(f\"Archivo descargado en {targetdir}\")\n",
    "            return out\n",
    "        except Exception as err:\n",
    "            print(f\"Error al descargar el archivo {pdbfullname}: {err}\", file=sys.stderr)\n",
    "            return None\n",
    "            \n",
    "download_pdb(template)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9d5a7c1e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archivo PDB limpio guardado como: ../run/modelamineto_dinamica/4zwb_clean.pdb\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'../run/modelamineto_dinamica/4zwb_clean.pdb'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#=== Limpieza de prefijos y residuos no deseados en un archivo PDB ===\n",
    "\n",
    "class CustomPDBSelector(Select):\n",
    "    \"\"\"\n",
    "    Permite seleccionar una cadena específica y excluir residuos HETATM que no sean agua (HOH).\n",
    "    \"\"\"\n",
    "    def __init__(self, chain_id):\n",
    "        self.chain_id = chain_id\n",
    "\n",
    "    def accept_chain(self, chain):\n",
    "    #   Acepta solo la cadena con el ID especificado.\n",
    "        return chain.id == self.chain_id\n",
    "\n",
    "    def accept_residue(self, residue):\n",
    "    #    Acepta residuos de tipo ATOM y residuos HETATM que sean agua (HOH).\n",
    "    #    Excluye otros residuos HETATM.\n",
    "    \n",
    "        # Si es un residuo HETATM y no es HOH (agua), no lo acepta.\n",
    "        if residue.id[0] != ' ' and residue.get_resname().strip() != 'HOH':\n",
    "            return False\n",
    "        return True # Acepta otros residuos (ATOM o HOH)\n",
    "\n",
    "def clean_pdb(template, chain_id='A'):\n",
    "    \"\"\"\n",
    "    Limpia un archivo PDB:\n",
    "    1. Selecciona una cadena específica.\n",
    "    2. Elimina residuos HETATM que no sean agua.\n",
    "    3. Filtra líneas que comienzan con prefijos no deseados (ANISOU, HET, HETNAM, HETSYN, FORMUL, CONECT).\n",
    "    \"\"\"\n",
    "    pdb_file = f\"../run/modelamineto_dinamica/{template}.pdb\"\n",
    "    clean_file = f\"../run/modelamineto_dinamica/{template}_clean.pdb\"\n",
    "\n",
    "    # 1. Parsear el archivo PDB\n",
    "    parser = PDBParser(QUIET=True)\n",
    "    try:\n",
    "        structure = parser.get_structure('structure', pdb_file)\n",
    "    except Exception as e:\n",
    "        print(f\"Error al parsear el archivo PDB {pdb_file}: {e}\", file=sys.stderr)\n",
    "        return None\n",
    "\n",
    "    # Verificar si la estructura o el modelo existen\n",
    "    if not structure or not structure[0]:\n",
    "        print(f\"No se encontraron modelos o estructura en el archivo PDB {pdb_file}\", file=sys.stderr)\n",
    "        return None\n",
    "    \n",
    "    # Verificar si la cadena especificada existe en el primer modelo\n",
    "    target_model = structure[0]\n",
    "    chain_ids = [chain.id for chain in target_model]\n",
    "    if chain_id not in chain_ids:\n",
    "        print(f\"La cadena '{chain_id}' no se encontró en la estructura. Cadenas disponibles: {chain_ids}\", file=sys.stderr)\n",
    "        return None\n",
    "    \n",
    "    # 2. Seleccionar la cadena y excluir HETATM (no-agua), escribiendo a un buffer en memoria\n",
    "    io = PDBIO()\n",
    "    io.set_structure(structure) # Se establece la estructura completa en PDBIO\n",
    "    \n",
    "    # Usar StringIO para capturar la salida de PDBIO en memoria en lugar de un archivo en disco\n",
    "    pdb_string_io = StringIO()\n",
    "    io.save(pdb_string_io, CustomPDBSelector(chain_id)) # Se pasa el selector personalizado\n",
    "    pdb_string_io.seek(0) # Rebobinar el buffer al inicio para poder leerlo\n",
    "    \n",
    "    # 3. Limpiar líneas adicionales no deseadas del buffer en memoria y escribir al archivo final\n",
    "    \n",
    "    exclude_prefixes_line_filter = ['ANISOU', 'HET', 'HETNAM', 'HETSYN', 'FORMUL', 'CONECT']\n",
    "    \n",
    "    with open(clean_file, 'w') as f_out:\n",
    "        for line in pdb_string_io:\n",
    "            # Escribir la línea solo si no comienza con ninguno de los prefijos a excluir\n",
    "            if not any(line.startswith(prefix) for prefix in exclude_prefixes_line_filter):\n",
    "                f_out.write(line)\n",
    "\n",
    "    print(f\"Archivo PDB limpio guardado como: {clean_file}\")\n",
    "    return clean_file\n",
    "\n",
    "clean_pdb(template, chain_id='A')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1175014a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fragment for chain A:\n",
      "TQKVTPALIFAITVATIGSFQFGYNTGVINAPEKIIKEFITKTLTDKGNAPPSEVLLTSLWSLSVAIFSVGGMIGSFSVGLFVNRFGRRNSMLIVNLLAVTGGCFMGLCKVAKSVEMLILGRLVIGLFCGLCTGFVPMYIGEISPTALRGAFGTLNQLGIVVGILVAQIFGLEFILGSEELWPLLLGFTILPAILQSAALPFCPESPRFLLINRKEEENAKQILQRLWGTQDVSQDIQEMKDESARMSQEKQVTVLELFRVSSYRQPIIISIVLQLSQQLSGINAVFYYSTGIFKDAGVQEPIYATIGAGVVNTIFTVVSLFLVERAGRRTLHMIGLGGMAFCSTLMTVSLLLKDNYNGMSFVCIGAILVFVAFFEIGPGPIPWFIVAELFSQGPRPAAMAVAGCSNWTSNFLVGLLFPSAAHYLGAYVFIIFTGFLITFLAFTFFKVPETRGRTFEDITRAFEGQAH\n"
     ]
    }
   ],
   "source": [
    "parser = PDBParser(QUIET=True)\n",
    "structure = parser.get_structure(\"protein\", f'../run/modelamineto_dinamica/{template}_clean.pdb')\n",
    "ppb = PPBuilder()\n",
    "\n",
    "chain_id = 'A'\n",
    "full_sequence = \"\"\n",
    "\n",
    "for model in structure:\n",
    "    for chain in model:\n",
    "        if chain.id == chain_id:\n",
    "            for pp in ppb.build_peptides(chain):\n",
    "                frag_seq = pp.get_sequence()\n",
    "                print(f\"Fragment for chain {chain.id}:\")\n",
    "                print(frag_seq)\n",
    "                full_sequence += str(frag_seq)\n",
    "\n",
    "fasta_template = f\"../run/modelamineto_dinamica/{template}.fasta\"\n",
    "with open(fasta_template, \"w\") as f:\n",
    "    f.write(f\">{template}\\n\")\n",
    "    f.write(full_sequence + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "af96adc6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Archivo guardado en ../run/modelamineto_dinamica/alignment.fasta\n",
      "Alignment successful\n",
      "Alignment saved to mafft_output.fasta\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "rescale = 1\n",
      "All-to-all alignment.\n",
      "tbfast-pair (aa) Version 7.505\n",
      "alg=L, model=BLOSUM62, 2.00, -0.10, +0.10, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "rescale = 1\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP     1 /1 \n",
      "done.\n",
      "tbfast (aa) Version 7.505\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "rescale = 1\n",
      "\n",
      "    0 / 2\n",
      "Segment   1/  1    1- 493\n",
      "done 001-001-1  identical.   \n",
      "dvtditr (aa) Version 7.505\n",
      "alg=A, model=BLOSUM62, 1.53, -0.00, -0.00, noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#=== Concatenar archivos fasta ===\n",
    "fastaP11166 = f'../data/GTR1_HUMAN.fasta'\n",
    "output = '../run/modelamineto_dinamica/alignment.fasta'\n",
    "\n",
    "with open(output, 'w') as out:\n",
    "    for file in [fasta_template, fastaP11166]:\n",
    "        with open(file, 'r') as f:\n",
    "            out.write(f.read())\n",
    "\n",
    "print(f'Archivo guardado en {output}')\n",
    "\n",
    "#=== Alineamiento MAFFT ===\n",
    "mafft_cmd = [\"mafft\", \"--auto\", output]\n",
    "try:\n",
    "    alignment = subprocess.check_output(mafft_cmd, universal_newlines=True)\n",
    "    print(\"Alignment successful\")\n",
    "\n",
    "    # Save result\n",
    "    with open(\"../run/modelamineto_dinamica/outputmft.fasta\", \"w\") as ali_file:\n",
    "        ali_file.write(alignment)\n",
    "    print(\"Alignment saved to mafft_output.fasta\")\n",
    "\n",
    "except subprocess.CalledProcessError as e:\n",
    "    print(\"Alignment failed:\", e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "dadcab56",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done\n"
     ]
    }
   ],
   "source": [
    "# Read MAFFT output file\n",
    "with open(\"../run/modelamineto_dinamica/outputmft.fasta\", \"r\") as file:\n",
    "    lines = file.readlines()\n",
    "\n",
    "# Split sequences\n",
    "records = {}\n",
    "current_id = None\n",
    "for line in lines:\n",
    "    line = line.strip()\n",
    "    if line.startswith(\">\"):\n",
    "        current_id = line[1:]\n",
    "        records[current_id] = \"\"\n",
    "    else:\n",
    "        records[current_id] += line\n",
    "\n",
    "# Now format as PIR\n",
    "if len(records) != 2:\n",
    "    print(\"Expected exactly 2 sequences for PIR format.\")\n",
    "else:\n",
    "    seq_id, template_id = list(records.keys())\n",
    "    seq_seq = records[seq_id]\n",
    "    template_seq = records[template_id]\n",
    "\n",
    "    pir_lines = [\n",
    "        f\">P1;{uniprotcd}\",\n",
    "        f\"sequence:{uniprotcd}::A::A::::\",\n",
    "        template_seq + \"*\",\n",
    "        f\">P1;{seq_id}\",\n",
    "        f\"structureX:{seq_id}::A::A::::\",\n",
    "        seq_seq + \"*\"\n",
    "    ]\n",
    "\n",
    "    with open(\"../run/modelamineto_dinamica/input_mdller.ali\", \"w\") as f:\n",
    "        f.write(\"\\n\".join(pir_lines))\n",
    "        print(\"done\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "b18bb9ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Inputs: ['../run/modelamineto_dinamica/input_mdller.ali'], 4zwb, secuencia de P11166\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    49272    45808\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   796908\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -66521.8594\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -66521.859375\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/surf5.de\n",
      "Pair library             : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    66.667000\n",
      "Sequence length          :          492\n",
      "Compactness              :     0.121785\n",
      "Native energy (pair)     :  -433.374815\n",
      "Native energy (surface)  :    -4.724728\n",
      "Native energy (combined) :    -8.653687\n",
      "Z score (pair)           :    -8.841653\n",
      "Z score (surface)        :    -5.269231\n",
      "Z score (combined)       :    -9.170761\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     9110\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1751.9840\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    3902       0      0   0.005   0.005      28.402       1.000\n",
      " 2 Bond angle potential               :    5293       0     11   1.972   1.972      407.95       1.000\n",
      " 3 Stereochemical cosine torsion poten:    2538       0     22  41.883  41.883      675.92       1.000\n",
      " 4 Stereochemical improper torsion pot:    1630       0      0   1.176   1.176      48.700       1.000\n",
      " 5 Soft-sphere overlap restraints     :    9110       0      0   0.001   0.001      2.0305       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   10974       0      0   0.079   0.079      74.918       1.000\n",
      "10 Distance restraints 2 (N-O)        :   11111       0      0   0.105   0.105      147.77       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     491       0      3   4.041   4.041      94.535       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     412       0      1  64.858  64.858      53.322       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     301       0      2  66.150  66.150      62.395       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     122       0      0  68.074  68.074      67.318       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      37       0      0 102.062 102.062      23.453       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5462       0      0   0.305   0.305      100.74       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     490      24     23  15.402  38.516     -244.92       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3045       0      3   0.648   0.648      209.46       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: P11166.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   40144.5430\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13364   1M   2E C   N       7    9 -125.67 -117.80   23.11    0.98  -63.60  172.71   26.53\n",
      "    1          2E   2E N   CA      9   10  158.53  136.80                  -40.30\n",
      "    2  13366   3P   4S C   N      23   25  -62.24  -72.40   10.20    0.70  -64.10  173.57   12.47\n",
      "    2          4S   4S N   CA     25   26  151.43  152.40                  -35.00\n",
      "    3  13367   4S   5S C   N      29   31 -138.52 -136.60    9.41    0.43  -64.10 -179.37   18.40\n",
      "    3          5S   5S N   CA     31   32  160.41  151.20                  -35.00\n",
      "    4  13414  51R  52Y C   N     395  397  -96.74  -98.40   27.75    2.95  -63.50  147.88   21.86\n",
      "    4         52Y  52Y N   CA    397  398  100.70  128.40                  -43.40\n",
      "    5  13417  54E  55S C   N     420  422  -60.85  -72.40   11.75    0.77  -64.10  174.78   12.47\n",
      "    5         55S  55S N   CA    422  423  150.25  152.40                  -35.00\n",
      "    6  13541 178S 179I C   N    1343 1345  -82.89  -63.40   46.95    8.41  -63.40   46.95    8.41\n",
      "    6        179I 179I N   CA   1345 1346  -86.31  -43.60                  -43.60\n",
      "    7  13744 381V 382G C   N    2940 2942 -163.21  -62.40  101.42   18.53  -62.40  101.42   18.53\n",
      "    7        382G 382G N   CA   2942 2943  -52.35  -41.20                  -41.20\n",
      "    8  13835 472A 473S C   N    3657 3659  -41.38  -72.40   60.73    2.78  -64.10  137.08   11.44\n",
      "    8        473S 473S N   CA   3659 3660  100.19  152.40                  -35.00\n",
      "    9  13836 473S 474Q C   N    3663 3665 -177.82 -121.10   58.78    1.93  -63.80 -159.81   35.14\n",
      "    9        474Q 474Q N   CA   3665 3666  155.14  139.70                  -40.30\n",
      "   10  13837 474Q 475S C   N    3672 3674 -110.63 -136.60   46.89    1.70  -64.10  154.34    9.51\n",
      "   10        475S 475S N   CA   3674 3675  112.15  151.20                  -35.00\n",
      "   11  13839 476D 477K C   N    3686 3688 -104.81 -118.00   19.84    1.01  -62.90  170.51   24.62\n",
      "   11        477K 477K N   CA   3688 3689  153.92  139.10                  -40.80\n",
      "   12  13840 477K 478T C   N    3695 3697  -75.62  -78.10   21.76    0.85  -63.20  170.74   21.10\n",
      "   12        478T 478T N   CA   3697 3698  128.19  149.80                  -42.10\n",
      "   13  13842 479P 480E C   N    3709 3711  -60.59  -69.30    8.74    0.69  -63.60  176.53   23.38\n",
      "   13        480E 480E N   CA   3711 3712  143.19  142.50                  -40.30\n",
      "   14  13843 480E 481E C   N    3718 3720  -86.59  -69.30   39.40    3.51  -63.60  149.18   18.72\n",
      "   14        481E 481E N   CA   3720 3721  107.09  142.50                  -40.30\n",
      "   15  13844 481E 482L C   N    3727 3729 -135.92 -108.50   56.36    3.21  -63.50  144.00   17.65\n",
      "   15        482L 482L N   CA   3729 3730   83.26  132.50                  -41.20\n",
      "   16  13845 482L 483F C   N    3735 3737 -132.28 -124.20   33.19    1.73  -63.20  170.07   19.74\n",
      "   16        483F 483F N   CA   3737 3738  111.11  143.30                  -44.30\n",
      "   17  13846 483F 484H C   N    3746 3748 -157.14 -125.60   74.08    3.28  -63.20  147.77   16.11\n",
      "   17        484H 484H N   CA   3748 3749   71.77  138.80                  -42.30\n",
      "   18  13847 484H 485P C   N    3756 3758  -69.06  -58.70   10.67    1.47  -64.50  179.81   13.18\n",
      "   18        485P 485P N   CA   3758 3759  -33.04  -30.50                  147.20\n",
      "   19  13848 485P 486L C   N    3763 3765 -155.29 -108.50   52.81    2.33  -63.50 -173.96   31.48\n",
      "   19        486L 486L N   CA   3765 3766  156.99  132.50                  -41.20\n",
      "   20  13849 486L 487G C   N    3771 3773 -138.06 -167.20   53.04    1.47   82.20 -174.64   14.36\n",
      "   20        487G 487G N   CA   3773 3774  130.28  174.60                    8.50\n",
      "   21  13850 487G 488A C   N    3775 3777 -144.52 -134.00   10.56    0.39  -62.50 -168.50   35.78\n",
      "   21        488A 488A N   CA   3777 3778  146.05  147.00                  -40.90\n",
      "   22  13851 488A 489D C   N    3780 3782  -99.64  -96.50   15.65    0.65  -63.30  143.55   15.87\n",
      "   22        489D 489D N   CA   3782 3783   98.88  114.20                  -40.00\n",
      "   23  13852 489D 490S C   N    3788 3790 -140.91 -136.60    5.58    0.17  -64.10 -173.21   19.01\n",
      "   23        490S 490S N   CA   3790 3791  154.73  151.20                  -35.00\n",
      "   24  13853 490S 491Q C   N    3794 3796  -83.56  -73.00   16.16    0.93  -63.80  167.92   25.74\n",
      "   24        491Q 491Q N   CA   3796 3797  152.94  140.70                  -40.30\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   18   37  167  312  455  435  485  657  739  697\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_484W> Dihedral still outside +-90:      -90.0291\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   797813\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -66601.3594\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -66601.359375\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/surf5.de\n",
      "Pair library             : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    66.667000\n",
      "Sequence length          :          492\n",
      "Compactness              :     0.132466\n",
      "Native energy (pair)     :  -397.292023\n",
      "Native energy (surface)  :    -8.094019\n",
      "Native energy (combined) :    -8.582757\n",
      "Z score (pair)           :    -8.702614\n",
      "Z score (surface)        :    -5.698712\n",
      "Z score (combined)       :    -9.442342\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     9048\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1873.3330\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    3902       0      0   0.005   0.005      31.064       1.000\n",
      " 2 Bond angle potential               :    5293       0     14   2.045   2.045      435.58       1.000\n",
      " 3 Stereochemical cosine torsion poten:    2538       0     19  41.506  41.506      672.30       1.000\n",
      " 4 Stereochemical improper torsion pot:    1630       0      0   1.263   1.263      55.222       1.000\n",
      " 5 Soft-sphere overlap restraints     :    9048       0      0   0.002   0.002      3.7763       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   10974       0      0   0.091   0.091      95.428       1.000\n",
      "10 Distance restraints 2 (N-O)        :   11111       0      5   0.127   0.127      195.03       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     491       0      4   4.157   4.157      100.05       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     412       0      1  64.043  64.043      57.941       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     301       0      3  64.561  64.561      81.886       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     122       0      1  66.365  66.365      76.776       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      37       0      0  88.364  88.364      26.237       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5462       0      0   0.294   0.294      91.902       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     490      23     23  15.852  37.081     -236.18       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3045       0      5   0.615   0.615      186.31       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: P11166.V99990002\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   40704.4805\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13364   1M   2E C   N       7    9  -94.27 -117.80   29.15    0.87  -63.60  162.81   20.18\n",
      "    1          2E   2E N   CA      9   10  119.60  136.80                  -40.30\n",
      "    2  13367   4S   5S C   N      29   31  -67.36  -72.40   10.77    0.80  -64.10  163.12   12.01\n",
      "    2          5S   5S N   CA     31   32  161.91  152.40                  -35.00\n",
      "    3  13414  51R  52Y C   N     395  397  -98.04  -98.40   28.53    3.04  -63.50  147.38   21.72\n",
      "    3         52Y  52Y N   CA    397  398   99.87  128.40                  -43.40\n",
      "    4  13417  54E  55S C   N     420  422  -62.61  -72.40   12.14    0.62  -64.10  179.78   12.94\n",
      "    4         55S  55S N   CA    422  423  145.22  152.40                  -35.00\n",
      "    5  13420  57L  58P C   N     442  444  -69.22  -64.50   30.86    2.07  -58.70  152.17   13.44\n",
      "    5         58P  58P N   CA    444  445  177.69  147.20                  -30.50\n",
      "    6  13421  58P  59T C   N     449  451   78.15  -63.20  143.32   21.00  -63.20  143.32   21.00\n",
      "    6         59T  59T N   CA    451  452  -65.83  -42.10                  -42.10\n",
      "    7  13541 178S 179I C   N    1343 1345  -85.38  -63.40   42.90    7.82  -63.40   42.90    7.82\n",
      "    7        179I 179I N   CA   1345 1346  -80.45  -43.60                  -43.60\n",
      "    8  13744 381V 382G C   N    2940 2942 -163.35  -62.40  101.43   18.48  -62.40  101.43   18.48\n",
      "    8        382G 382G N   CA   2942 2943  -51.03  -41.20                  -41.20\n",
      "    9  13835 472A 473S C   N    3657 3659  -37.61  -72.40   65.66    3.00  -64.10  134.36   11.51\n",
      "    9        473S 473S N   CA   3659 3660   96.72  152.40                  -35.00\n",
      "   10  13836 473S 474Q C   N    3663 3665 -129.86 -121.10   22.92    0.91  -63.80  172.00   28.77\n",
      "   10        474Q 474Q N   CA   3665 3666  160.88  139.70                  -40.30\n",
      "   11  13837 474Q 475S C   N    3672 3674  -70.51  -72.40   34.69    2.11  -64.10  138.11   10.40\n",
      "   11        475S 475S N   CA   3674 3675 -172.96  152.40                  -35.00\n",
      "   12  13838 475S 476D C   N    3678 3680  -54.92  -70.90   16.05    1.16  -63.30  171.42   20.66\n",
      "   12        476D 476D N   CA   3680 3681  148.79  150.30                  -40.00\n",
      "   13  13839 476D 477K C   N    3686 3688  -65.66  -70.20   23.50    1.59  -62.90  158.17   20.39\n",
      "   13        477K 477K N   CA   3688 3689  117.34  140.40                  -40.80\n",
      "   14  13840 477K 478T C   N    3695 3697 -148.30 -124.80   36.56    1.15  -63.20  169.33   27.17\n",
      "   14        478T 478T N   CA   3697 3698  171.50  143.50                  -42.10\n",
      "   15  13842 479P 480E C   N    3709 3711  -41.91  -69.30   40.12    2.24  -63.60  155.02   21.88\n",
      "   15        480E 480E N   CA   3711 3712  113.19  142.50                  -40.30\n",
      "   16  13843 480E 481E C   N    3718 3720 -129.70 -117.80   16.59    0.53  -63.60 -176.35   28.22\n",
      "   16        481E 481E N   CA   3720 3721  148.36  136.80                  -40.30\n",
      "   17  13844 481E 482L C   N    3727 3729  -81.65  -70.70   11.59    0.91  -63.50  174.37   25.29\n",
      "   17        482L 482L N   CA   3729 3730  145.37  141.60                  -41.20\n",
      "   18  13845 482L 483F C   N    3735 3737 -125.42 -124.20   19.91    0.97  -63.20  178.91   21.18\n",
      "   18        483F 483F N   CA   3737 3738  123.43  143.30                  -44.30\n",
      "   19  13846 483F 484H C   N    3746 3748 -158.50 -125.60   34.00    1.01  -63.20 -164.82   30.56\n",
      "   19        484H 484H N   CA   3748 3749  147.37  138.80                  -42.30\n",
      "   20  13848 485P 486L C   N    3763 3765 -123.50 -108.50   15.75    0.70  -63.50 -171.68   23.48\n",
      "   20        486L 486L N   CA   3765 3766  137.31  132.50                  -41.20\n",
      "   21  13849 486L 487G C   N    3771 3773   63.10   78.70   18.65    0.58   82.20  165.49    8.99\n",
      "   21        487G 487G N   CA   3773 3774 -155.89 -166.10                    8.50\n",
      "   22  13850 487G 488A C   N    3775 3777  -74.85  -68.20   14.93    0.98  -62.50  160.91   26.99\n",
      "   22        488A 488A N   CA   3777 3778  158.67  145.30                  -40.90\n",
      "   23  13852 489D 490S C   N    3788 3790 -106.23 -136.60   32.09    1.04  -64.10 -179.20   11.31\n",
      "   23        490S 490S N   CA   3790 3791  140.82  151.20                  -35.00\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   14   26  148  325  434  419  514  676  731  697\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   793275\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -66842.3594\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -66842.359375\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/surf5.de\n",
      "Pair library             : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    66.667000\n",
      "Sequence length          :          492\n",
      "Compactness              :     0.069208\n",
      "Native energy (pair)     :  -425.333665\n",
      "Native energy (surface)  :    -5.716272\n",
      "Native energy (combined) :    -8.642133\n",
      "Z score (pair)           :    -8.592888\n",
      "Z score (surface)        :    -5.150204\n",
      "Z score (combined)       :    -9.070678\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     8989\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1783.1448\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    3902       0      0   0.005   0.005      27.746       1.000\n",
      " 2 Bond angle potential               :    5293       0      9   1.955   1.955      401.20       1.000\n",
      " 3 Stereochemical cosine torsion poten:    2538       0     19  41.849  41.849      677.83       1.000\n",
      " 4 Stereochemical improper torsion pot:    1630       0      0   1.182   1.182      49.084       1.000\n",
      " 5 Soft-sphere overlap restraints     :    8989       0      0   0.002   0.002      2.5730       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   10974       0      1   0.083   0.083      75.530       1.000\n",
      "10 Distance restraints 2 (N-O)        :   11111       0      3   0.116   0.116      165.95       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     491       0      5   3.991   3.991      92.228       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     412       0      1  66.353  66.353      51.155       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     301       0      1  64.974  64.974      64.077       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     122       0      0  67.230  67.230      67.980       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      37       0      0 102.249 102.249      27.886       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5462       0      0   0.288   0.288      90.228       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     490      28     22  17.075  41.917     -212.13       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3045       0      4   0.629   0.629      201.81       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: P11166.V99990003\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   40615.1367\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13364   1M   2E C   N       7    9  -75.52  -69.30   18.28    1.59  -63.60  166.04   21.51\n",
      "    1          2E   2E N   CA      9   10  125.31  142.50                  -40.30\n",
      "    2  13365   2E   3P C   N      16   18  -64.33  -58.70   11.38    1.39  -64.50  172.42   12.87\n",
      "    2          3P   3P N   CA     18   19  -40.39  -30.50                  147.20\n",
      "    3  13366   3P   4S C   N      23   25  163.44 -136.60   60.33    2.23  -64.10 -146.70   25.01\n",
      "    3          4S   4S N   CA     25   26  157.81  151.20                  -35.00\n",
      "    4  13367   4S   5S C   N      29   31  144.42 -136.60   80.24    2.83  -64.10 -139.94   26.81\n",
      "    4          5S   5S N   CA     31   32  165.37  151.20                  -35.00\n",
      "    5  13414  51R  52Y C   N     395  397  -95.83  -98.40   36.84    3.92  -63.50  138.87   20.48\n",
      "    5         52Y  52Y N   CA    397  398   91.65  128.40                  -43.40\n",
      "    6  13417  54E  55S C   N     420  422  -60.22  -72.40   12.72    0.79  -64.10  176.31   12.55\n",
      "    6         55S  55S N   CA    422  423  148.73  152.40                  -35.00\n",
      "    7  13541 178S 179I C   N    1343 1345  -81.82  -63.40   48.08    8.56  -63.40   48.08    8.56\n",
      "    7        179I 179I N   CA   1345 1346  -88.01  -43.60                  -43.60\n",
      "    8  13724 361L 362P C   N    2775 2777  -73.48  -64.50   29.45    1.80  -58.70  154.97   14.02\n",
      "    8        362P 362P N   CA   2777 2778  175.24  147.20                  -30.50\n",
      "    9  13725 362P 363W C   N    2782 2784   72.53  -63.00  136.24   23.28  -63.00  136.24   23.28\n",
      "    9        363W 363W N   CA   2784 2785  -30.29  -44.20                  -44.20\n",
      "   10  13744 381V 382G C   N    2940 2942 -163.99  -62.40  102.12   18.63  -62.40  102.12   18.63\n",
      "   10        382G 382G N   CA   2942 2943  -51.65  -41.20                  -41.20\n",
      "   11  13833 470G 471G C   N    3648 3650 -178.85   82.20  105.00    5.47  -62.40  117.36   20.32\n",
      "   11        471G 471G N   CA   3650 3651  -26.62    8.50                  -41.20\n",
      "   12  13835 472A 473S C   N    3657 3659  -53.78  -72.40   25.03    1.21  -64.10  170.98   13.03\n",
      "   12        473S 473S N   CA   3659 3660  135.67  152.40                  -35.00\n",
      "   13  13836 473S 474Q C   N    3663 3665 -134.25 -121.10   36.22    1.46  -63.80  162.33   27.55\n",
      "   13        474Q 474Q N   CA   3665 3666  173.45  139.70                  -40.30\n",
      "   14  13837 474Q 475S C   N    3672 3674 -114.23 -136.60   32.65    1.07  -64.10  169.97   10.48\n",
      "   14        475S 475S N   CA   3674 3675  127.41  151.20                  -35.00\n",
      "   15  13838 475S 476D C   N    3678 3680 -156.42  -96.50   61.59    2.51  -63.30  168.10   17.96\n",
      "   15        476D 476D N   CA   3680 3681   99.96  114.20                  -40.00\n",
      "   16  13839 476D 477K C   N    3686 3688  -75.49  -70.20    5.50    0.37  -62.90  177.77   23.79\n",
      "   16        477K 477K N   CA   3688 3689  141.88  140.40                  -40.80\n",
      "   17  13840 477K 478T C   N    3695 3697  -78.62  -78.10    3.16    0.15  -63.20  171.91   23.07\n",
      "   17        478T 478T N   CA   3697 3698  146.68  149.80                  -42.10\n",
      "   18  13842 479P 480E C   N    3709 3711 -123.51 -117.80    5.93    0.18  -63.60 -171.52   22.42\n",
      "   18        480E 480E N   CA   3711 3712  138.40  136.80                  -40.30\n",
      "   19  13843 480E 481E C   N    3718 3720 -102.51 -117.80   33.37    1.31  -63.60  152.49   18.48\n",
      "   19        481E 481E N   CA   3720 3721  107.14  136.80                  -40.30\n",
      "   20  13844 481E 482L C   N    3727 3729 -141.01 -108.50   32.57    1.57  -63.50 -171.55   23.16\n",
      "   20        482L 482L N   CA   3729 3730  130.57  132.50                  -41.20\n",
      "   21  13845 482L 483F C   N    3735 3737 -135.80 -124.20   14.50    0.78  -63.20 -166.93   22.63\n",
      "   21        483F 483F N   CA   3737 3738  134.60  143.30                  -44.30\n",
      "   22  13846 483F 484H C   N    3746 3748 -158.46 -125.60   32.98    1.10  -63.20 -159.73   31.21\n",
      "   22        484H 484H N   CA   3748 3749  141.53  138.80                  -42.30\n",
      "   23  13848 485P 486L C   N    3763 3765 -103.99 -108.50   36.91    1.94  -63.50  142.92   17.97\n",
      "   23        486L 486L N   CA   3765 3766   95.87  132.50                  -41.20\n",
      "   24  13849 486L 487G C   N    3771 3773   55.28   78.70   44.17    0.69   82.20  139.77    8.11\n",
      "   24        487G 487G N   CA   3773 3774 -128.65 -166.10                    8.50\n",
      "   25  13850 487G 488A C   N    3775 3777 -128.20 -134.00   14.61    0.90  -62.50  171.76   31.67\n",
      "   25        488A 488A N   CA   3777 3778  160.41  147.00                  -40.90\n",
      "   26  13851 488A 489D C   N    3780 3782  -61.98  -70.90   10.05    0.55  -63.30  174.33   21.44\n",
      "   26        489D 489D N   CA   3782 3783  145.68  150.30                  -40.00\n",
      "   27  13852 489D 490S C   N    3788 3790  -62.86  -72.40   15.52    0.71  -64.10  175.17   12.77\n",
      "   27        490S 490S N   CA   3790 3791  140.17  152.40                  -35.00\n",
      "   28  13853 490S 491Q C   N    3794 3796 -126.64 -121.10   10.65    0.39  -63.80 -177.90   30.11\n",
      "   28        491Q 491Q N   CA   3796 3797  148.79  139.70                  -40.30\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   14   37  160  304  481  408  489  652  703  743\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   795043\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -66581.0312\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -66581.031250\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/surf5.de\n",
      "Pair library             : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    66.667000\n",
      "Sequence length          :          492\n",
      "Compactness              :     0.145101\n",
      "Native energy (pair)     :  -425.548199\n",
      "Native energy (surface)  :    -8.556917\n",
      "Native energy (combined) :    -8.963658\n",
      "Z score (pair)           :    -8.786207\n",
      "Z score (surface)        :    -5.814379\n",
      "Z score (combined)       :    -9.597803\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     8957\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1765.3624\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    3902       0      0   0.005   0.005      28.593       1.000\n",
      " 2 Bond angle potential               :    5293       0      8   2.006   2.006      418.38       1.000\n",
      " 3 Stereochemical cosine torsion poten:    2538       0     21  41.740  41.740      677.33       1.000\n",
      " 4 Stereochemical improper torsion pot:    1630       0      0   1.242   1.242      53.449       1.000\n",
      " 5 Soft-sphere overlap restraints     :    8957       0      0   0.001   0.001      2.3277       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   10974       0      0   0.078   0.078      75.226       1.000\n",
      "10 Distance restraints 2 (N-O)        :   11111       0      0   0.107   0.107      153.81       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     491       0      4   4.083   4.083      96.553       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     412       0      1  66.343  66.343      65.490       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     301       0      0  63.218  63.218      61.118       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     122       0      0  78.284  78.284      74.523       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      37       0      0  85.998  85.998      25.220       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5462       0      0   0.301   0.301      97.204       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     490      22     24  17.043  38.494     -238.72       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3045       0      1   0.596   0.596      174.85       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: P11166.V99990004\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   40245.3008\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13364   1M   2E C   N       7    9  -67.12  -69.30    2.29    0.20  -63.60  176.52   23.73\n",
      "    1          2E   2E N   CA      9   10  143.22  142.50                  -40.30\n",
      "    2  13366   3P   4S C   N      23   25   66.58  -72.40  139.52    9.60  -64.10 -141.52   25.26\n",
      "    2          4S   4S N   CA     25   26  140.09  152.40                  -35.00\n",
      "    3  13414  51R  52Y C   N     395  397 -101.26  -98.40   31.87    3.36  -63.50  145.06   21.22\n",
      "    3         52Y  52Y N   CA    397  398   96.66  128.40                  -43.40\n",
      "    4  13417  54E  55S C   N     420  422  -64.12  -72.40    8.53    0.54  -64.10  174.65   12.65\n",
      "    4         55S  55S N   CA    422  423  150.35  152.40                  -35.00\n",
      "    5  13541 178S 179I C   N    1343 1345  -82.41  -63.40   47.39    8.46  -63.40   47.39    8.46\n",
      "    5        179I 179I N   CA   1345 1346  -87.02  -43.60                  -43.60\n",
      "    6  13744 381V 382G C   N    2940 2942 -163.25  -62.40  101.47   18.54  -62.40  101.47   18.54\n",
      "    6        382G 382G N   CA   2942 2943  -52.46  -41.20                  -41.20\n",
      "    7  13835 472A 473S C   N    3657 3659  -61.56  -72.40   39.96    2.01  -64.10  148.97   10.95\n",
      "    7        473S 473S N   CA   3659 3660  113.94  152.40                  -35.00\n",
      "    8  13836 473S 474Q C   N    3663 3665  179.33 -121.10   60.07    2.10  -63.80 -151.81   36.48\n",
      "    8        474Q 474Q N   CA   3665 3666  147.41  139.70                  -40.30\n",
      "    9  13837 474Q 475S C   N    3672 3674 -163.00 -136.60   26.43    1.08  -64.10 -158.94   21.69\n",
      "    9        475S 475S N   CA   3674 3675  149.94  151.20                  -35.00\n",
      "   10  13839 476D 477K C   N    3686 3688  -88.89  -70.20   21.68    1.34  -62.90  169.81   23.55\n",
      "   10        477K 477K N   CA   3688 3689  151.39  140.40                  -40.80\n",
      "   11  13840 477K 478T C   N    3695 3697 -115.87 -124.80   11.51    0.39  -63.20 -174.05   20.82\n",
      "   11        478T 478T N   CA   3697 3698  136.23  143.50                  -42.10\n",
      "   12  13842 479P 480E C   N    3709 3711  -66.33  -69.30   12.16    0.82  -63.60  171.03   22.66\n",
      "   12        480E 480E N   CA   3711 3712  130.71  142.50                  -40.30\n",
      "   13  13843 480E 481E C   N    3718 3720 -137.13 -117.80   19.37    0.68  -63.60 -167.00   22.63\n",
      "   13        481E 481E N   CA   3720 3721  138.14  136.80                  -40.30\n",
      "   14  13844 481E 482L C   N    3727 3729 -107.90 -108.50   64.50    3.48  -63.50  117.89   14.55\n",
      "   14        482L 482L N   CA   3729 3730   68.01  132.50                  -41.20\n",
      "   15  13845 482L 483F C   N    3735 3737 -149.98 -124.20   26.73    0.84  -63.20 -173.28   31.47\n",
      "   15        483F 483F N   CA   3737 3738  150.37  143.30                  -44.30\n",
      "   16  13846 483F 484H C   N    3746 3748   67.78   56.30  129.20   12.42  -63.20 -162.21   21.78\n",
      "   16        484H 484H N   CA   3748 3749  169.49   40.80                  -42.30\n",
      "   17  13848 485P 486L C   N    3763 3765  -67.86  -70.70    2.92    0.28  -63.50  176.54   24.76\n",
      "   17        486L 486L N   CA   3765 3766  142.32  141.60                  -41.20\n",
      "   18  13849 486L 487G C   N    3771 3773  100.55   78.70   50.09    0.74   82.20  141.52    7.78\n",
      "   18        487G 487G N   CA   3773 3774  148.82 -166.10                    8.50\n",
      "   19  13850 487G 488A C   N    3775 3777 -144.95 -134.00   13.89    0.36  -62.50 -176.84   34.43\n",
      "   19        488A 488A N   CA   3777 3778  155.55  147.00                  -40.90\n",
      "   20  13851 488A 489D C   N    3780 3782  -97.98  -96.50   19.99    0.84  -63.30  177.56   20.06\n",
      "   20        489D 489D N   CA   3782 3783  134.14  114.20                  -40.00\n",
      "   21  13852 489D 490S C   N    3788 3790 -146.19 -136.60   29.82    1.28  -64.10  167.12   18.02\n",
      "   21        490S 490S N   CA   3790 3791  179.43  151.20                  -35.00\n",
      "   22  13853 490S 491Q C   N    3794 3796  -62.98  -73.00   13.55    0.79  -63.80  171.89   25.32\n",
      "   22        491Q 491Q N   CA   3796 3797  131.58  140.70                  -40.30\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   17   31  161  330  439  413  482  685  700  739\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      ">> Model assessment by DOPE potential\n",
      "iatmcls_286W> MODEL atom not classified:  VAL:OXT  VAL\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):   793531\n",
      "Dynamic pairs routine                             : 1, NATM x NATM double loop\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        1    9999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :   15.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       F       F       F       T\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :      -66751.7344\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "DOPE score               : -66751.734375\n",
      ">> Model assessment by GA341 potential\n",
      "\n",
      "Surface library          : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/surf5.de\n",
      "Pair library             : /home/gonza/miniconda3/envs/modeller/lib/modeller-10.7/modlib/pair9.de\n",
      "Chain identifier         : _\n",
      "% sequence identity      :    66.667000\n",
      "Sequence length          :          492\n",
      "Compactness              :     0.116697\n",
      "Native energy (pair)     :  -420.841108\n",
      "Native energy (surface)  :    -4.243167\n",
      "Native energy (combined) :    -8.351127\n",
      "Z score (pair)           :    -8.901706\n",
      "Z score (surface)        :    -5.104417\n",
      "Z score (combined)       :    -9.203666\n",
      "GA341 score              :     1.000000\n",
      "\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      492\n",
      "Number of all, selected real atoms                :     3812    3812\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    45808   45808\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     8951\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        1789.9109\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    3902       0      0   0.005   0.005      29.059       1.000\n",
      " 2 Bond angle potential               :    5293       0     11   1.988   1.988      410.50       1.000\n",
      " 3 Stereochemical cosine torsion poten:    2538       0     17  41.383  41.383      664.27       1.000\n",
      " 4 Stereochemical improper torsion pot:    1630       0      0   1.195   1.195      50.619       1.000\n",
      " 5 Soft-sphere overlap restraints     :    8951       0      0   0.002   0.002      5.6591       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :   10974       0      1   0.085   0.085      82.567       1.000\n",
      "10 Distance restraints 2 (N-O)        :   11111       0      0   0.113   0.113      166.63       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     491       0      3   4.036   4.036      94.303       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     412       0      3  64.331  64.331      53.092       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:     301       0      2  66.488  66.488      70.754       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:     122       0      1  70.230  70.230      78.919       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      37       0      0 104.525 104.525      23.425       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    5462       0      0   0.284   0.284      87.280       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     490      24     19  15.705  39.133     -235.88       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :    3045       0      4   0.645   0.645      208.72       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: P11166.V99990005\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   40483.1875\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13364   1M   2E C   N       7    9  -72.60  -69.30    8.62    0.76  -63.60  175.07   22.86\n",
      "    1          2E   2E N   CA      9   10  134.54  142.50                  -40.30\n",
      "    2  13366   3P   4S C   N      23   25  -79.94  -72.40    8.86    0.47  -64.10  168.69   13.23\n",
      "    2          4S   4S N   CA     25   26  157.06  152.40                  -35.00\n",
      "    3  13367   4S   5S C   N      29   31  -92.66  -72.40   32.91    1.51  -64.10  149.41   12.74\n",
      "    3          5S   5S N   CA     31   32  178.34  152.40                  -35.00\n",
      "    4  13414  51R  52Y C   N     395  397  -98.28  -98.40   31.99    3.40  -63.50  144.07   21.19\n",
      "    4         52Y  52Y N   CA    397  398   96.41  128.40                  -43.40\n",
      "    5  13417  54E  55S C   N     420  422  -59.34  -72.40   13.43    0.86  -64.10  175.79   12.46\n",
      "    5         55S  55S N   CA    422  423  149.27  152.40                  -35.00\n",
      "    6  13541 178S 179I C   N    1343 1345  -82.27  -63.40   47.83    8.53  -63.40   47.83    8.53\n",
      "    6        179I 179I N   CA   1345 1346  -87.54  -43.60                  -43.60\n",
      "    7  13744 381V 382G C   N    2940 2942 -163.19  -62.40  101.37   18.51  -62.40  101.37   18.51\n",
      "    7        382G 382G N   CA   2942 2943  -52.07  -41.20                  -41.20\n",
      "    8  13833 470G 471G C   N    3648 3650  179.73   82.20  102.15    5.52  -62.40  119.44   20.47\n",
      "    8        471G 471G N   CA   3650 3651  -21.90    8.50                  -41.20\n",
      "    9  13835 472A 473S C   N    3657 3659  -63.66  -72.40   12.05    0.58  -64.10  179.11   13.00\n",
      "    9        473S 473S N   CA   3659 3660  144.11  152.40                  -35.00\n",
      "   10  13836 473S 474Q C   N    3663 3665 -130.14 -121.10   20.96    0.81  -63.80  174.22   29.11\n",
      "   10        474Q 474Q N   CA   3665 3666  158.60  139.70                  -40.30\n",
      "   11  13837 474Q 475S C   N    3672 3674 -128.81 -136.60    9.37    0.29  -64.10 -169.66   18.34\n",
      "   11        475S 475S N   CA   3674 3675  145.99  151.20                  -35.00\n",
      "   12  13838 475S 476D C   N    3678 3680  -87.21  -70.90   42.39    1.45  -63.30  132.74   17.91\n",
      "   12        476D 476D N   CA   3680 3681 -170.57  150.30                  -40.00\n",
      "   13  13839 476D 477K C   N    3686 3688  -80.51  -70.20   84.11    6.33  -62.90   99.29   12.04\n",
      "   13        477K 477K N   CA   3688 3689   56.92  140.40                  -40.80\n",
      "   14  13840 477K 478T C   N    3695 3697 -141.62 -124.80   18.94    1.14  -63.20 -166.51   20.79\n",
      "   14        478T 478T N   CA   3697 3698  134.78  143.50                  -42.10\n",
      "   15  13842 479P 480E C   N    3709 3711 -153.70 -117.80   36.00    1.39  -63.60 -163.72   22.74\n",
      "   15        480E 480E N   CA   3711 3712  134.08  136.80                  -40.30\n",
      "   16  13843 480E 481E C   N    3718 3720 -155.40 -117.80   39.88    1.20  -63.60 -167.13   30.81\n",
      "   16        481E 481E N   CA   3720 3721  150.08  136.80                  -40.30\n",
      "   17  13844 481E 482L C   N    3727 3729 -105.23 -108.50   29.63    1.56  -63.50  150.17   18.91\n",
      "   17        482L 482L N   CA   3729 3730  103.06  132.50                  -41.20\n",
      "   18  13845 482L 483F C   N    3735 3737 -163.99 -124.20   45.09    1.21  -63.20 -178.28   31.45\n",
      "   18        483F 483F N   CA   3737 3738  164.50  143.30                  -44.30\n",
      "   19  13846 483F 484H C   N    3746 3748  -63.30  -67.60    7.92    0.49  -63.20  175.65   21.95\n",
      "   19        484H 484H N   CA   3748 3749  133.35  140.00                  -42.30\n",
      "   20  13848 485P 486L C   N    3763 3765 -122.20 -108.50   36.48    1.80  -63.50  163.40   26.32\n",
      "   20        486L 486L N   CA   3765 3766  166.31  132.50                  -41.20\n",
      "   21  13850 487G 488A C   N    3775 3777 -138.44 -134.00   14.86    0.72  -62.50  175.23   32.78\n",
      "   21        488A 488A N   CA   3777 3778  161.18  147.00                  -40.90\n",
      "   22  13851 488A 489D C   N    3780 3782  -64.85  -70.90   14.90    0.51  -63.30  176.68   21.71\n",
      "   22        489D 489D N   CA   3782 3783  136.68  150.30                  -40.00\n",
      "   23  13852 489D 490S C   N    3788 3790  -78.57  -72.40    7.50    0.60  -64.10  177.45   13.77\n",
      "   23        490S 490S N   CA   3790 3791  148.14  152.40                  -35.00\n",
      "   24  13853 490S 491Q C   N    3794 3796  -98.83 -121.10   29.88    1.51  -63.80  163.87   25.97\n",
      "   24        491Q 491Q N   CA   3796 3797  159.62  139.70                  -40.30\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0   16   29  162  295  452  413  487  660  702  714\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf     DOPE score    GA341 score\n",
      "----------------------------------------------------------------------\n",
      "P11166.B99990001.pdb          1751.98401   -66521.85938        1.00000\n",
      "P11166.B99990002.pdb          1873.33301   -66601.35938        1.00000\n",
      "P11166.B99990003.pdb          1783.14478   -66842.35938        1.00000\n",
      "P11166.B99990004.pdb          1765.36243   -66581.03125        1.00000\n",
      "P11166.B99990005.pdb          1789.91089   -66751.73438        1.00000\n",
      "\n"
     ]
    }
   ],
   "source": [
    "def autoModel_modeller():\n",
    "    alignments = []\n",
    "    alignments.append(\"../run/modelamineto_dinamica/input_mdller.ali\")\n",
    "\n",
    "    print(f'Inputs: {alignments}, {template}, secuencia de {uniprotcd}')\n",
    "    \n",
    "    for align in alignments:\n",
    "        env = Environ()\n",
    "        env.io.atom_files_directory = ['./', '../run/modelamineto_dinamica/']\n",
    "        a = AutoModel(env, alnfile=align,\n",
    "                knowns=f'{template}', sequence=uniprotcd,\n",
    "                assess_methods=(assess.DOPE,\n",
    "                                assess.GA341),)\n",
    "        a.starting_model = 1\n",
    "        a.ending_model = 5\n",
    "        a.make()\n",
    "        alignments = []\n",
    "\n",
    "autoModel_modeller()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "17c5e1d8",
   "metadata": {},
   "source": [
    "Ambiente miniconda glutoso (ambertools / openMM)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa657708",
   "metadata": {},
   "outputs": [],
   "source": [
    "#=== Ejecutar de manera local ===\n",
    "packmol-memgen -p P11166.pdb --log log_file --salt --salt_c Na+ --saltcon 0.15 \\\n",
    "-l POPC:POPE:POPI:POPS:POPA:SSM:CHL1//POPC:POPE:SSM:CHL1 -r 18:26:5:11:1:10:29//38:6:22:37 \\\n",
    "--ffprot ff19SB --fflip lipid21 --ffwat tip3p --dims 140 140 140 --pdb2pqr --pdb2pqr_pH 7.4 --nottrim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8468c0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "#=== Simulación de dinámica molecular 400 ps ===\n",
    "import openmm as mm\n",
    "import openmm.app as app\n",
    "import openmm.unit as unit\n",
    "from sys import stdout\n",
    "# load AMBER format files\n",
    "prmtop = app.AmberPrmtopFile('bilayer_P11166_DB01914_lipid.top')\n",
    "inpcrd = app.AmberInpcrdFile('bilayer_P11166_DB01914_lipid.crd')\n",
    "\n",
    "system = prmtop.createSystem(nonbondedMethod=app.PME, nonbondedCutoff=1*unit.nanometer,\n",
    "        constraints=app.HBonds)\n",
    "integrator = mm.LangevinMiddleIntegrator(300*unit.kelvin, 1/unit.picosecond, 0.002*unit.picoseconds)\n",
    "\n",
    "# create simulation instance.\n",
    "platform = mm.Platform.getPlatform('OpenCL')\n",
    "\n",
    "simulation = app.Simulation(prmtop.topology, system, integrator, platform)\n",
    "simulation.context.setPositions(inpcrd.positions)\n",
    "if inpcrd.boxVectors is not None:\n",
    "    simulation.context.setPeriodicBoxVectors(*inpcrd.boxVectors)\n",
    "\n",
    "simulation.minimizeEnergy()\n",
    "\n",
    "#Warm up systems.  Slowly warm up temperature - every 1000 steps raise the temperature by 5 K, ending at 300 K\n",
    "simulation.reporters.append(\n",
    "    app.StateDataReporter(\n",
    "        stdout,\n",
    "        5000,\n",
    "        step=True,\n",
    "        potentialEnergy=True,\n",
    "        temperature=True,\n",
    "        volume=True,\n",
    "        density=True\n",
    "    )\n",
    ")\n",
    "simulation.context.setVelocitiesToTemperature(5*unit.kelvin)\n",
    "print('Warming up the system...')\n",
    "T = 5\n",
    "mdsteps = 50000\n",
    "stages = 60\n",
    "steps_per_stage = int(mdsteps//stages)\n",
    "for i in range(60):\n",
    "    temperature = (T+(i*T))*unit.kelvin \n",
    "    if i%20 == 0:\n",
    "        print(f\"heating to {temperature}\")\n",
    "    integrator.setTemperature(temperature)\n",
    "    simulation.step(steps_per_stage)\n",
    "# give some extra steps to maintain temperature equilibrium\n",
    "print(f\"Now it's around 300K\")\n",
    "simulation.step(5001)\n",
    "\n",
    "#NPT equilibration, reducing backbone constraints\n",
    "mdsteps = 50000\n",
    "barostat = system.addForce(mm.MonteCarloBarostat(1*unit.atmosphere, 300*unit.kelvin))\n",
    "simulation.context.reinitialize(True)\n",
    "print('Running NPT equilibration...')\n",
    "simulation.step(mdsteps)\n",
    "simulation.saveState(\"equil.xml\")\n",
    "\n",
    "simulation.reporters[-1] = app.StateDataReporter(\n",
    "        stdout,\n",
    "        5000,\n",
    "        step=True,\n",
    "        potentialEnergy=True,\n",
    "        temperature=True,\n",
    "        volume=True,\n",
    "        density=True,\n",
    "        progress=True,\n",
    "        remainingTime=True,\n",
    "        speed=True,\n",
    "        totalSteps=mdsteps,\n",
    "        separator=\"\\t\"\n",
    "    )\n",
    "simulation.reporters.append(\n",
    "    app.DCDReporter(('10ns_traj.dcd'), 5000))\n",
    "mdsteps = 100000\n",
    "print('Running Production...')\n",
    "simulation.step(mdsteps)\n",
    "print('Done!')\n",
    "\n",
    "simulation.saveState(\"prod.xml\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "modeller",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
